src/main/java/domain/AbsenceHistory.java

```diff
@@ -0,0 +1,40 @@
+package domain;
+
+import dto.AbsenceResultDto;
+import dto.AttendanceResultDto;
+import java.util.List;
+
+public class AbsenceHistory {
+
+    private final List<AttendanceResultDto> attendanceResultDtos;
+
+    public AbsenceHistory(final List<AttendanceResultDto> attendanceResultDtos) {
+        this.attendanceResultDtos = attendanceResultDtos;
+    }
+
+    public AbsenceResultDto calculate() {
+        int attendance = attendanceCalculate();
+        int lateness = lateCalculate();
+        int absence = absenceCalculate();
+
+        String absenceStatus = AbsencePolicy.getAbsencePolicy(absence, lateness);
+        return new AbsenceResultDto(attendance, lateness, absence, absenceStatus);
+    }
+
+    private int lateCalculate() {
+        return calculateAbsence(AttendanceState.LATENESS);
+    }
+
+    private int absenceCalculate() {
+        return calculateAbsence(AttendanceState.ABSENCE);
+    }
+
+    private int attendanceCalculate() {
+        return calculateAbsence(AttendanceState.ATTENDANCE);
+    }
+
+    private int calculateAbsence(AttendanceState state) {
+        return (int) attendanceResultDtos.stream().filter(dto -> dto.attendanceState().equals(state.getDescription()))
```

Comment:
filter 안의 비교 로직을 보니 항상 실패할 것으로 보이는데, 확인 부탁드려요

Reply:
헉 확인하지 못했네요😭
```suggestion
return (int) attendanceResultDtos.stream()
                .filter(dto ->
                        dto.attendanceState().getDescription().equals(state.getDescription()))
```
이렇게 고칠 수 있을 것 같습니다!

Reply:
아래와 같이 변경해주셨네요ㅎ 문제는 해소되었으나, 값을 꺼내어 비교하고 있네요. getter의 사용을 자제하는 규칙에 위배하고 있어요.
```java
    private int calculateAbsence(AttendanceState state) {
        return (int) attendanceResultDtos.stream()
                .filter(dto ->
                        dto.attendanceState().equals(state))
                .count();
    }
```

Reply:
getter를 사용하지 않기 위해서 dto 에서 검증 로직을 넣어줘야 된다고 생각합니다! 그렇다면 데이터 전송만을 위한 dto 객체에게 값 검증 로직과 같은 로직들을 넣어줘도 괜찮을지 궁금합니다!

Reply:
말씀 주셨듯 DTO에는 로직을 만들지 않아야 해요. 도기는 이 미션에서 DTO를 왜 사용하셨나요? 지금의 AttendanceResultDto은 어디의 변경에 가장 영향을 받나요? 그렇다면 그 변경은 어디까지 전파되나요?

Reply:
AttendanceResultDto를 만든 가장 큰 이유는 원하는 데이터를 저장하는 객체를 만들고 싶었기 때문입니다. 지금 사용하는 DTO의 의존성을 보면 Attendance 객체에서 가장 많이 의존하고 있습니다. 만약 DTO에서 변경이 일어난다면 Attendance 또한 수정을 해야한다고 생각해요.